ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Tooltips/FSharpIdentifierTooltipProvider.fs

```diff
@@ -27,6 +27,15 @@ type FSharpIdentifierTooltipProvider(lifetime, solution, presenter, xmlDocServic
         (RichText.Empty, text.GetFormattedParts()) ||> Seq.fold (fun result part ->
             result.Append(part.Text.Replace("<", "&lt;").Replace(">", "&gt;").Replace("\n", "<br>"), part.Style))
 
+    //todo: remove explicit formatting with HTML
+    let asDefinition (text: RichText) = "<div class='definition'><pre style='word-wrap: break-word; white-space: pre-wrap;'>" + (richTextEscapeToHtml text) + "</pre></div>"
+
+    let asContent (text: RichText) = "<div class='content'>" + (richTextEscapeToHtml text) + "</div>"
+
+    let createToolTip (header: RichText) (body: RichText) =
+        if body.IsEmpty then asContent header else
+        (asDefinition header).Append(body)
```

Comment:
Maybe add extension members instead of these functions? (Ideally outside the language support plugin.)
It'd look much cleaner.

Reply:
These are temporary crutches that we will not need at all when the platform will support the HTML rendering we need.

Reply:
FYI now that you're using `white-space: pre-wrap`, you can get rid of the hack I added so that spaces render correctly at the start of a line. See here: https://github.com/JetBrains/resharper-fsharp/blob/dbea151f684b3de6ba9fa5106988d3c2156efe36/ReSharper.FSharp/src/FSharp.Common/src/Util/FcsTaggedText.fs#L61-L62

The `'\xA0'` can be replaced with `' '`

Reply:
@DedSec256 Have you checked this?

Reply:
@auduchinok, I will look at this in more detail shortly